### PR TITLE
Parsing ingredients

### PR DESCRIPTION
**Pull request by nsafai**
Currently parsing and rendering ingredients on grocery list page (/cart/grocery-list)
image

**HOW TO TEST**

1. npm i
2. nodemon
3. Login
4. Add recipes to cart
5. Go to  /cart/grocery-list

**NOTES**
Note that the parsing is not 100% perfect but it works for 90% of ingredients. Future improvements that I suggest upon refactoring:

- [x] create an IngredientLine class, that has properties like quantity, unitOfMeasurement, description, yield, pointerToRecipe (this pointer will be useful if user wants to update yield of a specific recipe)

**What is not finished** (could be merged without doing these, to be discussed)
- [ ] polish grocery list view
- [ ] save when users check off a box, see TODO comment in `grocery-list.hbs` or don't use checkboxes for now
- [ ] combining similar ingredients and/or sorting of any kind other than by recipe

### CODE REVIEW

**Comment by PhyllisWong** [comment]
Great work Nico! I noticed that the PR was requesting to merge the branch into the master, so I changed it to develop.

File: controllers/cart.js [hunk 1]
```diff
@@ -31,4 +31,65 @@ module.exports = (app) => {
...
+        } else if ( ingrWordsArr.length == 2 && isNumber(ingrWordsArr[0]) ) {
+          // this catches instances like "1 Lime"
+          let qty = ingrWordsArr[0];
+          let unit = ''
+          let desc = ingrWordsArr.slice(1, ingrWordsArr.length).join(' ');
+          listOfIngredients.push([qty, unit, desc]);
+
+        } else {
+          // almost every other ingredient will land in here
+          let qty = ingrWordsArr[0];
```
Comment (lesliekimm) [inline controllers/cart.js]: what happens if the ingredient is more than one word? ie. sliced apples

Comment (PhyllisWong) [reply controllers/cart.js]: We might need to do something a bit more like this psuedoCode ingrWordsArr[1:]

Comment (nsafai) [reply controllers/cart.js]: good catch @lesliekimm -- this is an issue with the code as written. It's definitely not perfect right now unfortunately.

Better solutions we could try:
- check if the first word isNumber() -- > i tried this but the library i tried using wasn't always giving me true for some fractions
- find the unit of measurement (tsp, tbsp, etc.) and then assume everything before it is `qty` and everything after it is the `desc`

Comment (lesliekimm) [reply controllers/cart.js]: something you can leave as a TODO and create another ticket for figuring out how to parse ingredients properly. that could be a ticket in itself. you will need to deal with a lot of edge cases!

Comment (PhyllisWong) [reply controllers/cart.js]: Done

Comment (lesliekimm) [reply controllers/cart.js]: can you link the ticket here for reference?

**Pull request merged**